### PR TITLE
265- Keystone Models

### PR DESCRIPTION
**Pull request by haitheredavid**
This starts to scaffold up a  the model structure for this application. I added some additional models that we might need to consider for better data organizations, but I'm not sure what level of detail we want to get to with building these models with keystone. 

The models created are 
- [x] Scenario Model
- [x] Condition Model  (obstructors)
- [x] Focus Model
- [x] View Model
- [x] Point Model

Additional
- [x] Result Option Model
- [x] Sasaki Object Interface (not sure if using interfaces is a good idea at this level)

Notion [Page](https://www.notion.so/sasaki-associates/Create-Scenario-Model-755601fe6e504baf83089e548c70627c?pvs=4)

### CODE REVIEW

File: viewto-frontend/src/models/Point.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+@model("viewto/Point")
+export class Point extends Model({
+    // not entirely sure this is needed since the points don't have an id
+    // but maybe this could be useful if we want to extract the points somehow
+    sasakiId: prop<string>(),
+    // location of the item from the cloud 
+    index: prop<number>(),
+    // indicates that it's been saved to the layout panel
+    saved: prop<boolean>(false),
```
Comment (ericyoungberg) [inline viewto-frontend/src/models/Point.tsx]: `saved: prop<boolean>(false).withSetter() // setSaved(true)`

File: viewto-frontend/src/models/Scenario.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    // get all the obstructing conditions from the views
+    @computedTree
+    get Conditions(): ConditionContext[] {
+        return this.views.map((x) => x.condition);
+    }
+
+    // volatile 
+    data: null;
```
Comment (ericyoungberg) [inline viewto-frontend/src/models/Scenario.tsx]:
```
@observable
data: any;

@action
setData(data) {
    this.data = data;
}
```

File: viewto-frontend/src/models/Scenario.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { FocusContext, ConditionContext } from './Context';
+
+@model("viewto/View")
+class Scenario extends Model({
+    // the information linked to the speckle project loaded
+    project: prop<Project>(),
+    // the list of views stored to apply the settings 
+    views: prop<View[]>(),
+}) {
+
```
Comment (ericyoungberg) [inline viewto-frontend/src/models/Scenario.tsx]:
```
onInit() {
   reaction(
       () => [this.project],
       () => {
            if (this.project.id && this.project.model && this.project.version && this.data === undefined) {
                      // get the data
           }
        }
   );
}
```

**Comment by haitheredavid** [comment]
updated with input from @ericyoungberg after reviewing

**Review by ericyoungberg — APPROVED** [review]

**Pull request merged**